### ISSUE TITLE
file drag and drop into a webview tag

### ISSUE THREAD

**Issue by lee99** [nwjs/nw.js#4653]
Hi,
I need some help !
I have try to use htmls api to drag file in order to upload files.
it works fine within a sigle page , but it can't work  when that page inside a webview tag.

Sigle page:**upload.html** in package
webview page:**webview.html** in package

all the code package like this.
[sample.zip](https://github.com/nwjs/nw.js/files/195143/sample.zip)
btw. the guide page no like [wiki_Dragging-files-into-page](https://github.com/nwjs/nw.js/wiki/Dragging-files-into-page)

**Comment by rogerwang** [comment]
What's  the version are you using?

**Comment by lee99** [comment]
thx,it appeared in nw13.* and nw14.*  all version.
I am no sure is it a bug.the sample like this.

![nw](https://cloud.githubusercontent.com/assets/2275619/14550146/c5d0d2a4-02f8-11e6-8403-e772c18142f5.jpg)

**Comment by lee99** [comment]
the same as https://bugs.chromium.org/p/chromium/issues/detail?id=412373

**Comment by lee99** [comment]
@ghostoy

**Comment by martinsuchan** [comment]
Is this issue really depending on Chromium bug 412373, or can it be fixed separately?  
The broken File Drag&Drop is quite a blocker for us right now.

**Comment by rogerwang** [comment]
Yeah it's the same bug. I'll see if we can do something for it soon.

**Comment by rogerwang** [comment]
btw, before we fix this, you may consider using iframe with 'nwfaketop' property rather than webview.

**Comment by martinsuchan** [comment]
@rogerwang Any idea if displaying pages in iframe with "nwfaketop" will work if the page uses [X-Frame-Options	DENY](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) security header? This web uses this header if you need to test it: https://www.michalspacek.cz/

**Comment by rogerwang** [comment]
We used to support bypassing that header. Will check it.

**Comment by martinsuchan** [comment]
@rogerwang hopefully this flag disables the X-Frame-Options only in the root frame and not in all iframes. It's a useful security feature.

**Comment by rogerwang** [comment]
@martinsuchan I just tried it and it seems to be working.

**Comment by rogerwang** [comment]
@lee99 I tried your sample and it's working for me.  The scripts in webview.html need to be removed.

**Comment by martinsuchan** [comment]
@rogerwang So I just asked and sadly we can't use iframe instead of webview because webview runs content on different process, that's what we want, and iframe cannot do that. Also there is huge difference in properties and events on webview and iframe... so back to the start.

**Comment by rogerwang** [comment]
Then see my previous comment about OP's sample.

**Comment by martinsuchan** [comment]
More progress, I've just tested [NWJS SDK v17.3](https://dl.nwjs.io/v0.17.3/nwjs-sdk-v0.17.3-win-x64.zip) and [NWJS SDK v14.7](https://dl.nwjs.io/v0.14.7/nwjs-sdk-v0.14.7-win-x64.zip) with the sample above and removed scripts, and noticed different behavior:  
In 17.3 it works as expected, the nested webview handles the file drop fine.
In 14.7 the nested webview handles the file but at the same time the parent frame tries to open the file as well <- this is unwanted  
So there was surely some kind of bug that was fixed between 14.7 and 17.3. Can this fix for drop in webview be backported to v14.x release or we need to update to v17.x?

**Comment by rogerwang** [comment]
I'll see it in 0.14 branch. btw, you may also try adding the event listeners in parent frame, but do not prevent its propagation.
